### PR TITLE
Adds securityContext and resources to initContainers for clusterAgent

### PR DESCRIPTION
**Pull request by joesteffee**
#### What this PR does / why we need it:

Datadog installs are not able to pass a kubeaudit scan due to missing limits and securityContexts on some of the init containers. This allows injection of those parameters

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/templates/_containers-init-linux.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {{- define "containers-init-linux" -}}
 - name: init-volume
+{{ include "generate-security-context" (dict "securityContext" .Values.agents.containers.initContainers.securityContext "targetSystem" .Values.targetSystem "seccomp" "" "kubeversion" .Capabilities.KubeVersion.Version) | indent 2 }}
```
Comment (fanny-jiang) [inline charts/datadog/templates/_containers-init-linux.yaml:3]: Adding a custom `securityContext` does not seem to be supported in GKE Autopilot. Can you please add a condition here to check if `.Values.providers.gke.autopilot` is not enabled?

```suggestion
{{- if not .Values.providers.gke.autopilot }}
{{ include "generate-security-context" (dict "securityContext" .Values.agents.containers.initContainers.securityContext "targetSystem" .Values.targetSystem "seccomp" "" "kubeversion" .Capabilities.KubeVersion.Version) | indent 2 }}
{{- end }}
```

File: charts/datadog/templates/_containers-init-linux.yaml [hunk 2]
```diff
@@ -12,6 +13,7 @@
   resources:
 {{ toYaml .Values.agents.containers.initContainers.resources | indent 4 }}
 - name: init-config
+{{ include "generate-security-context" (dict "securityContext" .Values.agents.containers.initContainers.securityContext "targetSystem" .Values.targetSystem "seccomp" "" "kubeversion" .Capabilities.KubeVersion.Version) | indent 2 }}
```
Comment (fanny-jiang) [inline charts/datadog/templates/_containers-init-linux.yaml:16]: Here as well

```suggestion
{{- if not .Values.providers.gke.autopilot }}
{{ include "generate-security-context" (dict "securityContext" .Values.agents.containers.initContainers.securityContext "targetSystem" .Values.targetSystem "seccomp" "" "kubeversion" .Capabilities.KubeVersion.Version) | indent 2 }}
{{- end }}
```

**Review by fanny-jiang — CHANGES_REQUESTED** [review]
Thanks for your contribution! I tested the change in a GKE Autopilot cluster and it seems that it does not support overwriting the default securityContext. I left comments on adding a condition to check for GKE Autopilot. It'd be nice to add a warning for this in the `charts/datadog/templates/NOTES.txt` [file](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/NOTES.txt#L225) in case the config is enabled in GKE Autopilot.

**Comment by fanny-jiang** [comment]
Looks great! One last thing, can you please run `make docs` and push the changes?

**Review by fanny-jiang — APPROVED** [review]

**Review by CharlyF — APPROVED** [review]

**Pull request merged**